### PR TITLE
[prometheus] upgrade prometheus-operator to version v0.44.x 

### PR DESCRIPTION
**Pull request by alejandroEsc**
**What type of PR is this?**

https://jira.d2iq.com/browse/D2IQ-73462
https://jira.d2iq.com/browse/D2IQ-73540

**What this PR does/ why we need it**:

Makes changes to allow upgrade of prometheus operator

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Prometheus-operator upgraded to v0.44.x
```

**Checklist**

* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

**Comment by alejandroEsc** [comment]
Note to self: update grafana and alertmanager versions.

File: addons/prometheus/prometheus.yaml [hunk 1]
```diff
@@ -10,11 +10,11 @@ metadata:
...
   annotations:
-    catalog.kubeaddons.mesosphere.io/addon-revision: "0.38.1-28"
-    appversion.kubeaddons.mesosphere.io/prometheus-operator: "0.38.1"
+    catalog.kubeaddons.mesosphere.io/addon-revision: "0.44.0-1"
+    appversion.kubeaddons.mesosphere.io/prometheus-operator: "0.44.0"
     appversion.kubeaddons.mesosphere.io/prometheus: "2.19.2"
-    appversion.kubeaddons.mesosphere.io/alertmanager: "0.20.0"
-    appversion.kubeaddons.mesosphere.io/grafana: "6.7.3"
+    appversion.kubeaddons.mesosphere.io/alertmanager: "0.21.0"
+    appversion.kubeaddons.mesosphere.io/grafana: "6.1.17"
```
Comment (gracedo) [inline addons/prometheus/prometheus.yaml:17]: weird. It looks like we downgraded but we must have had the wrong version of grafana before
image

File: addons/prometheus/prometheus.yaml [hunk 2]
```diff
@@ -230,13 +230,13 @@ spec:
                 scheme: http
         prometheusSpec:
           image:
-            tag: v2.19.2
+            tag: v2.22.1
```
Comment (gracedo) [inline addons/prometheus/prometheus.yaml:233]: we can probably omit this. the chart defaults the image to 2.22.1 so we don't need to overwrite it here unless we needed to upgrade. the risk of leaving it here is that we may accidentally overwrite a new default in the next upgrade

File: addons/prometheus/prometheus.yaml [hunk 3]
```diff
@@ -25,8 +25,8 @@ metadata:
     # The prometheus-operator chart from prior Konvoy releases can't be upgraded to 8.10.0.
```
Comment (gracedo) [inline addons/prometheus/prometheus.yaml:25]: we need to update the values sha on the line above to `825e9732efe486f372c12015b13db52faa7cfc42`

File: addons/prometheus/prometheus.yaml [hunk 4]
```diff
@@ -10,11 +10,11 @@ metadata:
     # on the cluster, this hack will trigger re-queue on Addons until one exists.
     kubeaddons.mesosphere.io/hack-requires-defaultstorageclass: "true"
   annotations:
-    catalog.kubeaddons.mesosphere.io/addon-revision: "0.38.1-28"
-    appversion.kubeaddons.mesosphere.io/prometheus-operator: "0.38.1"
+    catalog.kubeaddons.mesosphere.io/addon-revision: "0.44.0-1"
+    appversion.kubeaddons.mesosphere.io/prometheus-operator: "0.44.0"
     appversion.kubeaddons.mesosphere.io/prometheus: "2.19.2"
```
Comment (gracedo) [inline addons/prometheus/prometheus.yaml:15]:
```suggestion
    appversion.kubeaddons.mesosphere.io/prometheus: "2.22.1"
```

File: addons/prometheus/prometheus.yaml (hunk 1 above)
Comment (alejandroEsc) [reply addons/prometheus/prometheus.yaml:17]: yeah im not sure what that is about, but i assumed we never had the correct version?

File: addons/prometheus/prometheus.yaml [hunk 5]
```diff
@@ -229,14 +229,12 @@ spec:
                 interval: 30s
                 scheme: http
         prometheusSpec:
-          image:
-            tag: v2.19.2
           thanos:
             version: v0.8.1
```
Comment (gracedo) [inline addons/prometheus/prometheus.yaml:233]: TODO later .. but we should upgrade this old version of thanos. and looks like there are some new thanos capabilities in these newer prom charts that we should look into and utilize if relevant :)

**Review by gracedo — APPROVED** [review]

**Comment by gracedo** [comment]
`harness.go:73: ERROR: addon prometheus using the previous version '0.38.1-28' did not come up healthy`
🤔 there was an error installing the previous version (for upgrade test) which is strange because installation of this previous version passed tests in the past
`experimental_logging_hooks.go:125: pod prometheus-prometheus-kubeaddons-prom-prometheus-0 in namespace kubeaddons was in phase Pending when testing ended`

still digging in..

**Comment by gracedo** [comment]
`TestIstioGroup` is now getting run for some reason and those tests are failing so the prometheus group tests aren't even gotten to :/

**Review by s12chung — APPROVED** [review]

**Review by branden — APPROVED** [review]

**Comment by alejandroEsc** [comment]
hmm 1m to delete? 
```
addons.go:265: waiting 1m0s for previously existing addon prometheus to be deleted
18:26:15
      harness.go:73: ERROR: previous version of addon prometheus did not delete properly after 1m0s
18:28:15
      harness.go:73: ERROR: could not forward port to prometheus pod: pod prometheus-prometheus-kubeaddons-prom-prometheus-0 is not running, it's in phase Pending
```

**Comment by gracedo** [comment]
Seems to be failing consistently on that now.. do we need to increase that timeout? https://github.com/mesosphere/kubeaddons/blob/master/test/utils/addons.go#L36

**Comment by alejandroEsc** [comment]
@gracedo thank you for figuring out the api change!

**Pull request merged**